ts/components/NewRemindEmailValidationOverlay.tsx

```diff
@@ -0,0 +1,192 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import { View, SafeAreaView } from "react-native";
+import {
+  ButtonLink,
+  IOPictogramSizeScale,
+  Label,
+  Pictogram,
+  VSpacer
+} from "@pagopa/io-app-design-system";
+import { useNavigation } from "@react-navigation/native";
+import I18n from "../i18n";
+
+import {
+  profileLoadRequest,
+  startEmailValidation
+} from "../store/actions/profile";
+import {
+  isProfileEmailValidatedSelector,
+  profileEmailSelector
+} from "../store/reducers/profile";
+import { useIODispatch, useIOSelector } from "../store/hooks";
+import ROUTES from "../navigation/routes";
+import { IOStyles } from "./core/variables/IOStyles";
+import FooterWithButtons from "./ui/FooterWithButtons";
+import { IOToast } from "./Toast";
+import BaseScreenComponent from "./screens/BaseScreenComponent";
+
+const emailSentTimeout = 10000 as Millisecond; // 10 seconds
+const profilePolling = 5000 as Millisecond; // 5 seconds
+
+const EMPTY_EMAIL = "";
+const VALIDATION_ILLUSTRATION_WIDTH: IOPictogramSizeScale = 80;
+
+const NewRemindEmailValidationOverlay = () => {
+  const dispatch = useIODispatch();
+  const optionEmail = useIOSelector(profileEmailSelector);
+  const isEmailValidated = useIOSelector(isProfileEmailValidatedSelector);
+  const [isVerifyButtonDisabled, setIsVerifyButtonDisabled] = useState(false);
+  const timeout = useRef<number | undefined>();
+  const navigation = useNavigation();
+
+  const email = pipe(
+    optionEmail,
+    O.getOrElse(() => EMPTY_EMAIL)
+  );
+
+  const sendEmailValidation = useCallback(
+    () => dispatch(startEmailValidation.request()),
+    [dispatch]
+  );
+
+  const reloadProfile = useCallback(
+    () => dispatch(profileLoadRequest()),
+    [dispatch]
+  );
+
+  // function to locate the title of the button. If the email is validated and if it is not, whether the confirmation email was sent or not
+  const controlButtonTitle = () => {
+    if (isEmailValidated) {
+      return I18n.t("global.buttons.continue");
+    } else {
+      if (isVerifyButtonDisabled) {
+        return I18n.t("email.newvalidate.buttonlabelsent");
+      } else {
+        return I18n.t("email.newvalidate.buttonlabelsentagain");
+      }
+    }
+  };
+
+  // this function contol if the button is disabled. It is disabled if the email is sent and the timeout is active
+  const isButtonDisabled = () => {
+    if (isEmailValidated) {
+      return false;
+    } else {
+      return isVerifyButtonDisabled;
+    }
+  };
+
+  const handleSendEmailValidationButton = () => {
+    if (isEmailValidated) {
+      // if email is validated the user navigate tu prefereces screen
+      navigation.navigate(ROUTES.PROFILE_NAVIGATOR, {
+        screen: ROUTES.PROFILE_PREFERENCES_NOTIFICATIONS
+      });
+    } else {
+      // send email validation only if it exists
+      pipe(
+        optionEmail,
+        O.map(_ => {
+          sendEmailValidation();
+        })
+      );
```

Comment:
If the e-mail is empty, then we don't send the validation e-mail to the user, but we actually show the successful toast non the less. We do must handle the missing e-mail scenario providing a contextual error message or maybe disable the button if the e-mail is empty. On the contrary, If we are sure that when we reach this code the e-mail is validated, anyway we should tie the successful toast to a non empty e-mail and do nothing in the case of an empty e-mail. What do you think?

Reply:
When you get to that function, the check if the mail is validated is used to understand which of the screens is to be rendered. 
I expect that the mail is always validated from the previous screen, so I have not inserted any checks and/or additional banners because there should be no other cases

Reply:
I agree. But in this case I would put all the logic below inside the sampe `map` block in the `pipe`. Under `sendEmailValidation`.

Reply:
done!